### PR TITLE
Fixed issues with phantom connected devices

### LINKED ISSUE #15642: Canisters rupturing while hooked to connectors can break the connector
(WEB REPORT BY: panthid REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)
ef8d0b5c9f957d2e7a7596677d590a9ce07a0414

>General description of the issue
Canisters rupturing while hooked to connectors can break the connector. This makes the connector unusable even if unwrenched and wrenched again.

>What you expected to happen
Connector to reset to unconnected state when the canister ruptures.

>What actually happened
Connector gets stuck and thinks there is a canister still connected and so can't accept a new canister.

>Steps to reproduce if possible
Rupture a canister while it is hooked to a connector. In this case it was an oxygen canister which had been connected to the cryo in box which had the vacuum pump left on for a while. The canister was ruptured with a cyborg circular saw.

### PR DESCRIPTION
**Pull request by DamianX**
Fixes #15642

:cl:
 * bugfix: Moving or destroying an object wrenched on a connector port will now properly disconnect the two.

### CODE REVIEW

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -393,7 +393,8 @@
 
 	if(loc)
 		loc.Exited(src)
-
+		for(var/atom/movable/AM in loc)
```
Comment (PJB3005) [inline code/game/atoms_movable.dm]: This would call `uncrossed` on `src` too, which probably isn't ideal.

Comment (DamianX) [reply code/game/atoms_movable.dm]: Should I just do `loc - src`?

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -390,7 +390,7 @@
 
 // harderforce is for things like lighting overlays which should only be moved in EXTREMELY specific sitations.
 /atom/movable/proc/forceMove(atom/destination,var/no_tp=0, var/harderforce = FALSE)
-	var/old_loc = loc
+	var/turf/old_loc = loc
```
Comment (Exxion) [inline code/game/atoms_movable.dm]: Why?

Comment (DamianX) [reply code/game/atoms_movable.dm]: Because https://travis-ci.org/vgstation-coders/vgstation13/builds/279050728

Comment (DamianX) [reply code/game/atoms_movable.dm]: Fuck, that should be atom, not turf.

**Comment by DamianX** [comment]
Tested the latest commit and it works.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**